src/cmd/kubectl.rs

```diff
@@ -1017,6 +1017,88 @@ where
     )
 }
 
+/// kubectl_get_crash_looping_pods: gets crash looping pods.
+///
+/// Arguments
+///
+/// * `kubernetes_config`: kubernetes config file path.
+/// * `namespace`: namespace to look into, if None, will look into all namespaces.
+/// * `selector`: selector to look for, if None, will look for anything.
+/// * `restarted_min_count`: minimum restart counts to be considered as crash looping. If None, default is 10.
+/// * `envs`: environment variables to be passed to kubectl.
+pub fn kubectl_get_crash_looping_pods<P>(
+    kubernetes_config: P,
+    namespace: Option<&str>,
+    selector: Option<&str>,
+    restarted_min_count: Option<usize>,
+    envs: Vec<(&str, &str)>,
+) -> Result<Vec<KubernetesPod>, SimpleError>
+where
+    P: AsRef<Path>,
+{
+    let restarted_min = restarted_min_count.unwrap_or_else(|| 10usize);
+    let pods = kubectl_exec_get_pods(kubernetes_config, namespace, selector, envs)?;
+
+    Ok(pods
+        .items
+        .into_iter()
+        .filter(|e| {
+            e.status.phase != KubernetesPodStatusPhase::Running
+                && e.status.container_statuses.as_ref().is_some()
+                && e.status
+                    .container_statuses
+                    .as_ref()
+                    .expect("Cannot get container statuses")
+                    .iter()
+                    .any(|e| e.restart_count >= restarted_min)
+        })
+        .collect::<Vec<KubernetesPod>>())
+}
+
+/// kubectl_exec_delete_pod: allow to delete a k8s pod if exists.
+///
+/// Arguments
+///
+/// * `kubernetes_config`: kubernetes config file path.
+/// * `pod_namespace`: pod's namespace.
+/// * `pod_name`: pod's name.
+/// * `envs`: environment variables to be passed to kubectl.
+pub fn kubectl_exec_delete_pod<P>(
+    kubernetes_config: P,
+    pod_namespace: &str,
+    pod_name: &str,
+    envs: Vec<(&str, &str)>,
+) -> Result<KubernetesPod, SimpleError>
+where
+    P: AsRef<Path>,
+{
+    let pod_to_be_deleted =
+        match kubectl_exec_get_pods(&kubernetes_config, Some(pod_namespace), Some(pod_name), envs.clone()) {
+            Ok(pods) => {
+                if pods.items.is_empty() {
+                    return Err(SimpleError::new(
+                        SimpleErrorKind::Other,
+                        Some(format!(
+                            "Cannot delete pod `{}` in namespace `{}`, pod is ot found.",
+                            pod_name, pod_namespace
+                        )),
+                    ));
+                }
+
+                pods.items[0].clone()
+            }
+            Err(e) => return Err(e),
+        };
+
+    kubectl_exec(
+        vec!["delete", "pod", pod_to_be_deleted.metadata.name.as_str()],
```

Comment:
If i remember well you have specify namespace and add an arg for pod name:

delete pod -n <namespace> --name <pod_name>

Reply:
Indeed, thanks, --name is not an option, but namespace is required.